doc/languages-frameworks/rust.section.md

```diff
@@ -74,6 +117,15 @@ pkgs.rustPlatform.buildRustPackage {
 }
 ```
 
+This is useful to avoid rebuilding Rust tools, since they are actually target
+agnostic and don't need to be rebuilt. But in the future, we should always
+build the Rust tools and standard library crates separately so there is no
+reason not to take the `stdenv.hostPlatform.rustc`-modifying approach, and the
+ad-hoc escape hatch to `buildRustPackage` can be removed.
+
+Note that currently custom targets aren't compiled with `std`, so `cargo test`
+will fail. This can be ignored by adding `doCheck = false;` to your derivation.
```

Comment:
Is having the user explicitly disable tests better than silently disabling them?

Reply:
Just to make sure, which tests are you talking about? Testing if rustc works, or the rust test crate?

Reply:
This refers to the checkPhase of buildRustPackage, where `cargo test` fails to run on users' code:

```
λ nix-shell . -A rustPackageWithCustomSysroot
$ cargo build # success
$ cargo test  # error
```